mk-modules.nix

```
@@ -0,0 +1,86 @@
+{ flakeInputs, mkModulesOpts }: let
+  lib = flakeInputs.nixpkgs.lib;
+
+  systems = mkModulesOpts.systems or [
+    "aarch64-darwin"
+    "aarch64-linux"
+    "x86_64-darwin"
+    "x86_64-linux"
+  ];
+
+  flakeSchema.options = {
+    apps = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      default = {};
+    };
+
+    checks = lib.mkOption {
+      type = lib.types.attrsOf lib.types.package;
+      default = {};
+    };
+
+    devShells = lib.mkOption {
+      type = lib.types.attrsOf lib.types.package;
+      default = {};
+    };
+
+    packages = lib.mkOption {
+      type = lib.types.attrsOf lib.types.package;
+      default = {};
+    };
+
+    nixosConfigurations = lib.mkOption {
+      # This type will be checked by lib.nixosSystem below
+      type = lib.types.attrsOf lib.types.unspecified;
+      default = {};
+    };
+  };
+
+  evalModules = system: lib.evalModules {
+    specialArgs = {
+      pkgs = import flakeInputs.nixpkgs { inherit system; };
+      inherit system;
+    };
+    modules = [
+      flakeSchema
+      (mkModulesOpts.config or {})
+    ] ++ (mkModulesOpts.modules or []);
```

Why is the argument to `mkModules` not just a module itself? Couldn't we pass in e.g:
```nix
{
    imports = [
      inputs.rust-module.garnixModules.default
    ];
    config = {
      rust.backend = {
        src = ./sample-rust-project;
        serverCommand = "sample-rust-project";
      };
    };
  }
```

And then here we say:

```suggestion
      mkModulesOpts
    ];
```

---

I considered this, (and originally wanted to call `modules` `config` for this reason), but thinking about it more I think it's nice to be able to take in non-module specific options here (like we currently do for `systems`).